### PR TITLE
[Feature] Propagate the digital signature infos to the UI

### PR DESCRIPTION
**Pull request by marcomaddy1982**
## Dependencies

JIRA:  https://wearezeta.atlassian.net/browse/ZIOS-12764
sync-engine:  https://github.com/wireapp/wire-ios-sync-engine/pull/1199
ui: https://github.com/wireapp/wire-ios/pull/4236

### CODE REVIEW

File: Source/Model/Message/File/SignatureStatus.swift [hunk 1]
```diff
@@ -18,13 +18,22 @@
 
 import Foundation
 
+// MARK: - SignatureObserver
+@objc(ZMSignatureObserver)
+public protocol SignatureObserver: NSObjectProtocol {
+    func willReceiveSignatureURL()
+    func didReceiveSignatureURL(_ url: URL)
+    func signatureAvailable(_ signature: Data) //FIX ME: type of the file
```
Comment (johnxnguyen) [inline Source/Model/Message/File/SignatureStatus.swift]: Is the fix me still relevant?

File: Source/Model/Message/File/SignatureStatus.swift [hunk 2]
```diff
@@ -35,16 +44,19 @@ public enum PDFSigningState: Int {
 
 public final class SignatureStatus : NSObject {
     
+    // MARK: - Private Property
     private(set) var asset: ZMAsset?
-    private(set) var managedObjectContext: NSManagedObjectContext?
+    private(set) var managedObjectContext: NSManagedObjectContext
 
+    // MARK: - Private Property
```
Comment (johnxnguyen) [inline Source/Model/Message/File/SignatureStatus.swift]: I think you mean public properties.

File: Source/Model/Message/File/SignatureStatus.swift [hunk 3]
```diff
@@ -55,29 +67,72 @@ public final class SignatureStatus : NSObject {
...
+                    case let .consentURLReceived(consentURL):
+                        observer?.didReceiveSignatureURL(consentURL)
+                }
+            }
+        }
+    }
+}
+
+// MARK: - DigitallySignatureNotification
+public enum DigitallySignatureNotificationState {
```
Comment (johnxnguyen) [inline Source/Model/Message/File/SignatureStatus.swift]: Did you mean `DigitalSignatureNotificationState`?

Comment (johnxnguyen) [inline Source/Model/Message/File/SignatureStatus.swift]: Also, perhaps this should be declared within the scope of of the notification class?

File: Source/Model/Message/File/SignatureStatus.swift (hunk 1 above)
Comment (marcomaddy1982) [reply Source/Model/Message/File/SignatureStatus.swift]: No no. I can remove that fix me.

File: Source/Model/Message/File/SignatureStatus.swift [hunk 4]
```diff
@@ -55,29 +67,77 @@ public final class SignatureStatus : NSObject {
...
+            }
+        }
+    }
+}
+
+// MARK: - DigitalSignatureNotification
+public class DigitalSignatureNotification: NSObject  {
+    
+    // MARK: - DigitalSignatureNotificationState
+    public enum DigitalSignatureNotificationState {
```
Comment (typfel) [inline Source/Model/Message/File/SignatureStatus.swift]: Since it's defined inside `DigitalSignatureNotification` it could be renamed to just `State`

**Pull request merged**